libs/langchain/src/agents/middlewareAgent/middleware/hitl.ts

```diff
@@ -1,33 +1,118 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { z } from "zod/v3";
-import { v4 as uuid } from "uuid";
 import { AIMessage, ToolMessage } from "@langchain/core/messages";
 import { interrupt } from "@langchain/langgraph";
 
 import { createMiddleware } from "../middleware.js";
-import type { ToolCall } from "../types.js";
+
+const ToolConfigSchema = z.object({
+  allowAccept: z.boolean().optional(),
+  allowEdit: z.boolean().optional(),
+  allowRespond: z.boolean().optional(),
+  description: z.string().optional(),
+});
+
+type ToolConfigSchema = z.input<typeof ToolConfigSchema>;
+type ToolCall = NonNullable<AIMessage["tool_calls"]>[number];
 
 /**
- * Interrupt request for tool approval
+ * Configuration that defines what actions are allowed for a human interrupt.
```

Comment:
Would be good to clarify the doc-string. Actions appear in two places and mean two different things:

1. There's the action requested by the agent / workflow etc
2. There's the action performed by the reviewer

It would be good to specify that this is the action performed by the request specifically otherwise users will think it's referring to the ActionRequest